code-build.js

```diff
@@ -50,7 +53,37 @@ async function waitForBuildEndTime(sdk, { id, logs }, nextToken) {
       cloudWatchLogs
         .getLogEvents({ logGroupName, logStreamName, startFromHead, nextToken })
         .promise()
-  ]);
+  ]).catch(err => {
+    errorDetected = true;
+    errMessage = err;
+
+    return [];
+  });
+
+  if (errorDetected) {
+    //We caught an error in trying to make the AWS api call, and are now checking to see if it was just a rate limiting error
+    if (
+      errMessage.message &&
+      errMessage.message.search("Rate exceeded") !== -1
+    ) {
+      //We were rate-limited, so add 15 seconds to the wait time
+      let newWait = wait + 15000;
```

Comment:
Use the `backOff` value here.
This will also make it _posible_ to have exponential backoff or speed up.
Also, tests will be fast and nice...